agents/core.py

```diff
@@ -31,6 +32,12 @@ def step(self):
         self.task_step()
         self.count += 1
 
+    def parse_chat(self):
```

Comment:
why do we need this here?  sorry I don't understand, these are supposed to go in .perceive()